aiida/cmdline/commands/cmd_daemon.py

```diff
@@ -271,84 +268,13 @@ def start_circus(foreground, number):
 
     .. note:: this should not be called directly from the commandline!
     """
-    from circus import get_arbiter
-    from circus import logger as circus_logger
-    from circus.circusd import daemonize
-    from circus.pidfile import Pidfile
-    from circus.util import check_future_exception_and_log, configure_logger
-
     from aiida.engine.daemon.client import get_daemon_client
+    get_daemon_client()._start_daemon(number_workers=number, foreground=foreground)  # pylint: disable=protected-access
 
-    if foreground and number > 1:
-        raise click.ClickException('can only run a single worker when running in the foreground')
-
-    client = get_daemon_client()
-
-    loglevel = client.loglevel
-    logoutput = '-'
-
-    if not foreground:
-        logoutput = client.circus_log_file
-
-    arbiter_config = {
-        'controller': client.get_controller_endpoint(),
-        'pubsub_endpoint': client.get_pubsub_endpoint(),
-        'stats_endpoint': client.get_stats_endpoint(),
-        'logoutput': logoutput,
-        'loglevel': loglevel,
-        'debug': False,
-        'statsd': True,
-        'pidfile': client.circus_pid_file,
-        'watchers': [{
-            'cmd': client.cmd_string,
-            'name': client.daemon_name,
-            'numprocesses': number,
-            'virtualenv': client.virtualenv,
-            'copy_env': True,
-            'stdout_stream': {
-                'class': 'FileStream',
-                'filename': client.daemon_log_file,
-            },
-            'stderr_stream': {
-                'class': 'FileStream',
-                'filename': client.daemon_log_file,
-            },
-            'env': get_env_with_venv_bin(),
-        }]
-    }  # yapf: disable
-
-    if not foreground:
-        daemonize()
-
-    arbiter = get_arbiter(**arbiter_config)
-    pidfile = Pidfile(arbiter.pidfile)
 
-    try:
-        pidfile.create(os.getpid())
-    except RuntimeError as exception:
-        echo.echo_critical(str(exception))
-
-    # Configure the logger
-    loggerconfig = None
-    loggerconfig = loggerconfig or arbiter.loggerconfig or None
-    configure_logger(circus_logger, loglevel, logoutput, loggerconfig)
-
-    # Main loop
-    should_restart = True
-
-    while should_restart:
-        try:
-            future = arbiter.start()
-            should_restart = False
-            if check_future_exception_and_log(future) is None:
-                should_restart = arbiter._restarting  # pylint: disable=protected-access
-        except Exception as exception:
-            # Emergency stop
-            arbiter.loop.run_sync(arbiter._emergency_stop)  # pylint: disable=protected-access
-            raise exception
-        except KeyboardInterrupt:
-            pass
-        finally:
-            arbiter = None
-            if pidfile is not None:
-                pidfile.unlink()
+@verdi_daemon.command('worker')
```

Comment:
I understand that you feel this makes more sense as a location - at the same time, have you ever encountered a use case for this that was not related to AiiDA development?

If so, perhaps this use case should be documented in the aiida docs. If not, I guess we could also leave this command in `verdi devel`

Reply:
It's not so much for developing, but can be useful for debugging if you want to run a single daemon worker in the foreground. I would keep it under `verdi daemon` as it centralizes all daemon code, especially since circus also calls `verdi daemon worker` to spawn a new worker. It is a bit weird to have the actual daemon call `verdi devel run_daemon` as a daemon worker.